cli/src/config.rs

```diff
@@ -290,10 +312,9 @@ impl ConfigEnv {
     /// Returns the paths to the existing user-specific config files or
     /// directories.
     pub fn existing_user_config_paths(&self) -> impl Iterator<Item = &Path> {
-        self.user_config_paths.iter().filter_map(|p| match p {
-            ConfigPath::Existing(path) => Some(path.as_path()),
-            _ => None,
-        })
+        self.user_config_paths
+            .iter()
+            .filter_map(|p| if p.exists() { Some(p.as_path()) } else { None })
```

Comment:
nit: `.filter(|p| p.exists()).map(|p| p.as_path())`

Reply:
is there a reason to do that instead of a `FilterMap`?

Reply:
I think `.filter(p).map(f)` more directly expresses the intent. If there were a function that returns `Option<&Path>`, I would use `.filter_map()`.